services/121-service/src/program-financial-service-provider-configurations/program-financial-service-provider-configurations.service.ts

```diff
@@ -1,124 +1,427 @@
+    }
+  }
+
+  public async updateProperty({
+    programId,
+    programFspConfigurationName,
+    propertyName,
+    property,
+  }: {
+    programId: number;
+    programFspConfigurationName: string;
+    propertyName: FinancialServiceProviderConfigurationProperties;
+    property: UpdateProgramFinancialServiceProviderConfigurationPropertyDto;
+  }): Promise<ProgramFinancialServiceProviderConfigurationPropertyResponseDto> {
+    // Find the configuration
+
+    const config = await this.getProgramFspConfigurationOrThrow(
+      programId,
+      programFspConfigurationName,
+    );
+    const existingProperty =
+      await this.getProgramFspConfigurationPropertyOrThrow(
+        config.id,
+        propertyName,
+      );
+
+    existingProperty.value =
+      ProgramFinancialServiceProviderConfigurationMapper.mapPropertyDtoValueToEntityValue(
+        property.value,
+        existingProperty.name,
+      );
+
+    const savedProperty =
+      await this.programFspConfigurationPropertyRepository.save(
+        existingProperty,
+      );
+
+    return ProgramFinancialServiceProviderConfigurationMapper.mapPropertyEntityToDto(
+      savedProperty,
+    );
+  }
+
+  public async deleteProperty({
+    programId,
+    programFspConfigurationName,
+    propertyName,
+  }: {
+    programId: number;
+    programFspConfigurationName: string;
+    propertyName: FinancialServiceProviderConfigurationProperties;
+  }): Promise<void> {
+    const config = await this.getProgramFspConfigurationOrThrow(
+      programId,
+      programFspConfigurationName,
+    );
+    const existingProperty =
+      await this.getProgramFspConfigurationPropertyOrThrow(
+        config.id,
+        propertyName,
+      );
+    await this.programFspConfigurationPropertyRepository.delete({
+      id: existingProperty.id,
+    });
+  }
+
+  private async createProperties(
+    programFspConfigurationId: number,
+    inputProperties: CreateProgramFinancialServiceProviderConfigurationPropertyDto[],
+  ): Promise<ProgramFinancialServiceProviderConfigurationPropertyEntity[]> {
+    const propertiesToSave =
+      ProgramFinancialServiceProviderConfigurationMapper.mapPropertyDtosToEntities(
+        inputProperties,
+        programFspConfigurationId,
+      );
+    return this.programFspConfigurationPropertyRepository.save(
+      propertiesToSave,
+    );
+  }
+
+  private async overwriteProperties(
+    programFspConfigurationId: number,
+    properties: CreateProgramFinancialServiceProviderConfigurationPropertyDto[],
+  ): Promise<ProgramFinancialServiceProviderConfigurationPropertyEntity[]> {
+    // delete all properties
+    await this.programFspConfigurationPropertyRepository.delete({
+      programFinancialServiceProviderConfigurationId: Equal(
+        programFspConfigurationId,
+      ),
+    });
+    // create new properties
+    return await this.createProperties(programFspConfigurationId, properties);
+  }
+
+  private validateFspExists(fspName: string): void {
```

Comment:
Should this not happen in the Fsp Helper class then?

Reply:
Yes, not super sure. I put it there because I think only the service or controller should throw 'HttpErrors' and HttpErrors should not be 'hidden' in repos and helper functions. 

So you could also call the helper function there and than `catch` the normal error and throw it after with a HttpError. However that is much more of hassle/unclear in this case. I also think that in JavaScript it is better to LBYL (Look Before You Leap) instead of EAFP (Easier to Ask for Forgiveness than Permission) (which is common in Python). So it's better to not try/ catch if it is not really nice.

Reply:
Ah I figured something else i put  @IsEnum(FinancialServiceProviders) in the dto and now the whole function can be removed.